cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/default/ui/casLoginView.jsp

```diff
@@ -109,6 +109,19 @@
     <div class="sidebar-content">
         <p><spring:message code="screen.welcome.security" /></p>
 
+        <c:if test="${!empty pac4jUrls}">
+            <div id="list-providers">
+                <h3><spring:message code="screen.welcome.label.loginwith" /></h3>
+                <form>
+                    <ul>
```

Comment:
Good point. I finally used the name of the providers as I encountered two problems: 30 images at least would have been required to match all available providers and I wasn't sure to know what would make sense for SAML, OpenID Connect, CAS OAuth wrapped server...